File: order-service/src/main/java/com/intexsoft/webshop/orderservice/model/Detail.java
package com.intexsoft.webshop.orderservice.model;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;
import org.hibernate.proxy.HibernateProxy;

import java.math.BigDecimal;
import java.util.Objects;

@Entity
@Table(name = "detail")
@Getter
@Setter
@ToString
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Detail {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;
    @Column(name = "product_id", nullable = false)
    Long productId;
    @Column(name = "product_price", nullable = false)
    BigDecimal productPrice;
    @Column(name = "quantity", nullable = false)
    Short quantity;
    @ManyToOne(fetch = FetchType.LAZY)
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    Order order;

    @Override
    public final boolean equals(Object object) {
        if (this == object) return true;
        if (object == null) return false;
        Class<?> oEffectiveClass = object instanceof HibernateProxy ? ((HibernateProxy) object).getHibernateLazyInitializer().getPersistentClass() : object.getClass();
        Class<?> thisEffectiveClass = this instanceof HibernateProxy ? ((HibernateProxy) this).getHibernateLazyInitializer().getPersistentClass() : this.getClass();
        if (thisEffectiveClass != oEffectiveClass) return false;
        Detail detail = (Detail) object;
        return getId() != null && Objects.equals(getId(), detail.getId());
    }

    @Override
    public final int hashCode() {
        return this instanceof HibernateProxy ? ((HibernateProxy) this).getHibernateLazyInitializer().getPersistentClass().hashCode() : getClass().hashCode();
    }
}

File: order-service-kt/src/main/kotlin/com/intexsoft/webshop/orderservicekt/model/Detail.kt
package com.intexsoft.webshop.orderservicekt.model

import jakarta.persistence.*
import java.math.BigDecimal

@Entity
@Table(name = "detail")
class Detail(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0,
    @Column(name = "product_id", nullable = false)
    var productId: Long,
    @Column(name = "product_price", nullable = false)
    var productPrice: BigDecimal,
    @Column(name = "quantity", nullable = false)
    var quantity: Short,
    @ManyToOne(fetch = FetchType.LAZY)
    var order: Order?,
)